book/docs/verification/onchain/solidity-sdk.md

```diff
@@ -64,6 +64,14 @@ contract Fibonacci {
 
 ```
 
+The recommended workflow is to use the `ISP1Verifier` interface on the [`SP1VerifierGateway`](./contract-addresses.md),
+and the `SP1VerifierGateway` will automatically route your proof to the correct verifier.
+
+It’s also recommended to have your program’s verification key be upgradeable within your contract, in case you want to switch to a different version of SP1.
+
+Succinct also maintains the ability to freeze verifiers in the event of a security issue,
```

Comment:
```suggestion
Succinct maintains the ability to freeze verifiers on the canonical verifier gateway in the event of a security issue to prevent abuse, but verifier contract deployment is permissionless to enable customizable security configurations.
```